File: APE16 /ape16.java
import java.util.Scanner;

public class OperacionesMatrices {
    // Definición de las dimensiones globales para las matrices (2 filas y 3 columnas)
    static final int FILAS = 2;
    static final int COLUMNAS = 3;
    static Scanner scanner = new Scanner(System.in);

    /**
     * Solicita al usuario los valores para llenar una matriz de tamaño FILAS x COLUMNAS.
     */
    public static int[][] completar_matriz(char nombreMatriz) {
        int[][] matriz = new int[FILAS][COLUMNAS];
        System.out.println("Ingrese los datos de la matriz " + nombreMatriz + ":");
        
        // Recorre cada posición (fila y columna) para pedir el valor al usuario
        for (int i = 0; i < FILAS; i++) {
            for (int j = 0; j < COLUMNAS; j++) {
                System.out.print("Elemento [" + i + "][" + j + "]: ");
                matriz[i][j] = scanner.nextInt();
            }
        }
        return matriz;
    }

    /**
     * Suma dos matrices elemento por elemento.
     */
    public static int[][] suma_matriz(int[][] m1, int[][] m2) {
        int[][] resultado = new int[FILAS][COLUMNAS];
        for (int i = 0; i < FILAS; i++) {
            for (int j = 0; j < COLUMNAS; j++) {
                resultado[i][j] = m1[i][j] + m2[i][j];
            }
        }
        return resultado;
    }

    /**
     * Resta la segunda matriz de la primera elemento por elemento.
     */
    public static int[][] resta_matriz(int[][] m1, int[][] m2) {
        int[][] resultado = new int[FILAS][COLUMNAS];
        for (int i = 0; i < FILAS; i++) {
            for (int j = 0; j < COLUMNAS; j++) {
                resultado[i][j] = m1[i][j] - m2[i][j];
            }
        }
        return resultado;
    }

    /**
     * Realiza una multiplicación elemento a elemento
     */
    public static int[][] multiplicacion_matriz(int[][] m1, int[][] m2) {
        int[][] resultado = new int[FILAS][COLUMNAS];
        for (int i = 0; i < FILAS; i++) {
            for (int j = 0; j < COLUMNAS; j++) {
                resultado[i][j] = m1[i][j] * m2[i][j];
            }
        }
        return resultado;
    }

    /**
     * Imprime en pantalla la matriz resultante de una operación en formato tabular.
     */
    public static void mostrar_resultado(int[][] matriz, String operacion) {
        System.out.println("\nResultado de " + operacion + ":");
        for (int i = 0; i < FILAS; i++) {
            for (int j = 0; j < COLUMNAS; j++) {
                System.out.print(matriz[i][j] + "\t");
            }
            System.out.println();
        }
    }

    /**
     * Función principal que coordina la entrada de datos, operaciones y salida.
     */
    public static void main(String[] args) {
        // Se solicitan las matrices A y B al usuario
        int[][] A = completar_matriz('A');
        int[][] B = completar_matriz('B');

        // Se realizan las operaciones correspondientes
        int[][] res_suma = suma_matriz(A, B);
        int[][] res_resta = resta_matriz(A, B);
        int[][] res_mult = multiplicacion_matriz(A, B);

        // Se muestran los resultados en pantalla
        mostrar_resultado(res_suma, "Suma");
        mostrar_resultado(res_resta, "Resta");
        mostrar_resultado(res_mult, "Multiplicacion");
        
        scanner.close();
    }
}

File: APE16 /ape16.py
# Definición de las dimensiones globales para las matrices (2 filas y 3 columnas)
FILAS = 2
COLUMNAS = 3


def completar_matriz(filas_columnas):  # Solicita al usuario los valores para llenar una matriz de tamaño FILAS x COLUMNAS.
   
    # Crea una matriz con las dimensiones especificadas
    matriz = [[0 for _ in range(COLUMNAS)] for _ in range(FILAS)]
    print(f"Ingrese los datos de la matriz {filas_columnas}:")
    
    # Recorre cada posición (fila y columna) para pedir el valor al usuario
    for i in range(FILAS):
        for j in range(COLUMNAS):
            matriz[i][j] = int(input(f"Elemento [{i}][{j}]: "))
            
    return matriz


def suma_matriz(m1, m2):  # Suma dos matrices elemento por elemento.
   
    return [[m1[i][j] + m2[i][j] for j in range(COLUMNAS)] for i in range(FILAS)]


def resta_matriz(m1, m2): # Resta la segunda matriz de la primera elemento por elemento.
    
    return [[m1[i][j] - m2[i][j] for j in range(COLUMNAS)] for i in range(FILAS)]


def multiplicacion_matriz(m1, m2):
    """
    Nota: Realiza una multiplicación elemento a elemento (Hadamard),
    no una multiplicación matricial algebraica tradicional.
    """
    return [[m1[i][j] * m2[i][j] for j in range(COLUMNAS)] for i in range(FILAS)]


def mostrar_resultado(matriz, operacion):
    """Imprime en pantalla la matriz resultante de una operación en formato tabular."""
    print(f"\nResultado de {operacion}:")
    for i in range(FILAS):
        for j in range(COLUMNAS):
            print(matriz[i][j], end="\t")
        print()


def main(): # Función principal que coordina la entrada de datos, operaciones y salida.
    # Se solicitan las matrices A y B al usuario
    A = completar_matriz('A')
    B = completar_matriz('B')

    # Se realizan las operaciones correspondientes
    res_suma = suma_matriz(A, B)
    res_resta = resta_matriz(A, B)
    res_mult = multiplicacion_matriz(A, B)

    # Se muestran los resultados en pantalla
    mostrar_resultado(res_suma, "Suma")
    mostrar_resultado(res_resta, "Resta")
    mostrar_resultado(res_mult, "Multiplicacion")


if __name__ == "__main__":
    main()
